lib/cloudwatchlogger/client/aws_sdk/threaded.rb

```diff
@@ -33,9 +35,11 @@ def start_thread
       end
 
       class DeliveryThread < Thread
-
-        def initialize(credentials, log_group_name, log_stream_name, opts={})
-          @credentials, @log_group_name, @log_stream_name, @opts = credentials, log_group_name, log_stream_name, opts
+        def initialize(credentials, log_group_name, log_stream_name, opts = {})
+          @credentials = credentials
+          @log_group_name = log_group_name
+          @log_stream_name = log_stream_name
+          @opts = opts
```

Comment:
AFAIK, need to wait until we're done mutating `opts` (next two lines) before we store it on the class. Whether this is true or not, I'd like the code to read consistently so it's obvious we intend to be concerned with one object here.

Reply:
I do think `opts` should be mutated before assigning it off to the class.  Moving the mutation up is 👍